### ISSUE TITLE
Implement Unit Tests for SignUp Page

### ISSUE THREAD

**Issue by rubemfsv** [rubemfsv/clean-react-app#11]
This issue is focused on implementing unit tests for the SignUp Page within our project. Unit testing is a crucial aspect of software development that helps ensure the reliability and correctness of individual code units, such as components.

**Goals:**

- Develop a suite of unit tests that thoroughly assess the behavior and functionality of the SignUp page.
- Verify that user interactions, form submissions, and error handling within the SignUp page operate as intended.
- Ensure that any changes or updates to the SignUp page in the future do not introduce regressions.

**Tasks:**

1. Review the existing unit tests for the Login page to understand their structure and coverage.
2. Create unit tests to cover various aspects of the SignUp page, including rendering, form validation, input handling, and submission.
3. Implement test cases to assess the SignUp page's behavior in different scenarios, such as successful sign-up, validation errors, and edge cases.
4. Validate that user input is processed correctly and securely, taking into account data sanitization and validation rules.
5. Test error-handling mechanisms to ensure that the page provides clear and user-friendly feedback in case of issues.
6. Assess the accessibility of the SignUp page to ensure it complies with accessibility standards and guidelines.
7. Aim for comprehensive code coverage, ensuring that critical parts of the SignUp page are thoroughly tested.
8. 

**Guidelines:**

- Follow best practices for unit testing.
- Write clear and descriptive test cases that cover both typical and edge cases.
- Pay attention to code quality, readability, and maintainability when writing tests.

**Comment by ghoshvidip26** [comment]
Is this Hacktoberfest Accepted? If yes, can I contribute?

**Comment by rubemfsv** [comment]
Hey @Vidip-Ghosh! Yes, this is accepted at Hacktoberfest, but to count on Hacktoberfest you have to wait until October! I added the labels.

**Comment by ghoshvidip26** [comment]
Please assign me this one.

**Comment by rubemfsv** [comment]
@Vidip-Ghosh done! Don't forget to ensure commit messages are following the conventional commits :smile:

**Comment by ghoshvidip26** [comment]
@rubemfsv which react library do you use for testing?

**Comment by pawas108ram** [comment]
Is this issue still assigned if not i would like to take it up .

**Comment by ghoshvidip26** [comment]
> Is this issue still assigned if not i would like to take it up .

Its assigned

**Comment by rubemfsv** [comment]
@Vidip-Ghosh @testing-library/react and jest. Look at the package.json, and dont add new libs. Please, check the tasks. Review the existing unit tests for the Login page to understand their structure and coverage.

**Comment by ghoshvidip26** [comment]
> @Vidip-Ghosh @testing-library/react and jest. Look at the package.json, and dont add new libs. Please, check the tasks. Review the existing unit tests for the Login page to understand their structure and coverage.

Sure Thanks a lot

**Comment by ghoshvidip26** [comment]
<img width="992" alt="Screenshot 2023-10-02 at 19 59 15" src="https://github.com/rubemfsv/clean-react-app/assets/91741581/622464b9-f21d-42f8-8a26-086d096713be">
Which URL are we supposed to place here?

**Comment by rubemfsv** [comment]
@Vidip-Ghosh as it is a template, you should replace for your own URL. But as this task is for unit test, you don't need these urls... You should use spies and stubs for the tests. Please, review the existing unit tests for the Login page to understand their structure and coverage.

**Comment by rubemfsv** [comment]
@Vidip-Ghosh are you still working on this?

**Comment by ghoshvidip26** [comment]
Yes I am working. Sorry for the delay, due to some work it's getting late.

**Comment by ghoshvidip26** [comment]
@rubemfsv I have raised a PR.

**Comment by rubemfsv** [comment]
@Vidip-Ghosh I`ve put some improvements there!
